jerry-core/vm/vm.c

```diff
@@ -473,6 +506,7 @@ opfunc_call (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
   else
   {
     arguments_list_len = frame_ctx_p->byte_code_p[1];
+    byte_code_length++;
```

Comment:
byte_code_length = 2.

This somehow could be done by directly increasing the byte code, since it is not used afterwards. From the opcode you can get the length, which is only needed when debugger is enabled.

Reply:
Unfortunately I'm not sure that we can do it directly.

As I've applied your suggestion it caused an error in `debugger_do_step.js` while sending the backtrace info.

```diff
--- tests/debugger/do_step.expected     2018-10-26 15:50:52.387344484 +0200
+++ tests/debugger/do_step.out.ufKJQwcUB4       2018-11-21 11:21:23.083086725 +0100
@@ -10 +10 @@
-Frame 2: tests/debugger/do_step.js:32
+Frame 2: tests/debugger/do_step.js:33
@@ -15 +15 @@
-Frame 1: tests/debugger/do_step.js:32
+Frame 1: tests/debugger/do_step.js:33
```

Reason:
 - It tries to send to actual bytecode position [here](https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/debugger/debugger.c#L182), but this value is increased by the actual bytecode length.

So as I see we can't increase the actual byte_code pointer since the ecma_op_function_call (...) has not finished the real call operation.

Reply:
It looks like this is again something debugger specific. 

What about changing the code to:
`arguments_list_len = *(++frame_ctx_p->byte_code_p);`

This way we are not fully reading the byte code, so not moving to the next instruction.

Or replace `byte_code_length` to save `byte_code_p` locally:
`uint8_t *byte_code_p = frame_ctx_p->byte_code_p;`

And update it accordingly. 

Setting a value is faster than += operation, since that needs reading the original value.

Reply:
+1 for your second suggestion, it looks promising.